controllers/provisioners/eks/helpers.go

```diff
@@ -457,14 +458,47 @@ func (ctx *EksInstanceGroupContext) GetLabelList() []string {
 	return labelList
 }
 
-func (ctx *EksInstanceGroupContext) GetBootstrapArgs() string {
+func (ctx *EksInstanceGroupContext) GetComputedBootstrapOptions() *v1alpha1.BootstrapOptions {
 	var (
 		instanceGroup = ctx.GetInstanceGroup()
+		state         = ctx.GetDiscoveredState()
 		configuration = instanceGroup.GetEKSConfiguration()
 	)
+	if instanceGroup.GetAnnotations()[CustomNetworkingEnabledAnnotation] == "true" {
+		hostNetworkPods, err := strconv.ParseInt(instanceGroup.GetAnnotations()[CustomNetworkingHostPodsAnnotation], 10, 64)
+		if err != nil {
+			hostNetworkPods = 2 //Default on EKS. Kube-Proxy and AWS VPC CNI
+		}
+
+		instanceTypes := state.GetInstanceTypeInfo()
+		for _, instanceType := range instanceTypes {
```

Comment:
Wonder if you can simplify this logic a little bit by adding a utility method under awsprovider e.g.

```go
// This will remove loop & first if
networkInfo := awsprovider.GetInstanceTypeNetworkInfo(types, type)
maxPods := (networkInfo.MaximumNetworkInterfaces-1) * (networkInfo.Ipv4AddressesPerInterface-1) + hostNetworkPods
if configuration.BootstrapOptions == nil {
        // Can avoid else if there it returns
	return &v1alpha1.BootstrapOptions{
		MaxPods: maxPods,
	}
}
if bootstrapOptions.MaxPods == 0 {
	bootstrapOptions.MaxPods = maxPods
}
return &bootstrapOptions
```

WDYT?

Reply:
Yeah, totally agree !